src/HIVCohortFilter/HIVCohortFilter.jsx

```diff
@@ -0,0 +1,364 @@
+                      days_to_follow_up
+                      pregnancy_status
+                      hiv_status
+                      drug_used
+                      visit_date
+                      visit_number
+                      visit_type
+                      visit_name
+                      weight
+                      weight_percentage
+                      follow_up_id
+                      tint
+                    }
+                  }
+                }
+              }
+            }
+        `;
+
+    return this.performQuery(query);
+  }
+
+  setViralLoadFromUser(e) {
+    const viralLoadFromUser = e.target.value;
+    if (typeof viralLoadFromUser !== 'undefined'
+            && viralLoadFromUser !== '' && !isNaN(parseFloat(viralLoadFromUser))) {
+      this.setState({ viralLoadFromUser: parseFloat(viralLoadFromUser) });
+    }
+  }
+
+  setNumConsecutiveMonthsFromUser(e) {
+    const numConsecutiveMonthsFromUser = e.target.value;
+    if (typeof numConsecutiveMonthsFromUser !== 'undefined'
+            && numConsecutiveMonthsFromUser !== ''
+            && !isNaN(parseFloat(numConsecutiveMonthsFromUser))) {
+      this.setState({ numConsecutiveMonthsFromUser: parseFloat(numConsecutiveMonthsFromUser) });
+    }
+  }
+
+  performQuery(queryString) { // eslint-disable-line
+    return fetchWithCreds({
+      path: `${arrangerGraphqlPath}`,
+      body: JSON.stringify({
+        query: queryString,
+      }),
+      method: 'POST',
+    })
+      .then(
+        ({ status, data }) => data, // eslint-disable-line no-unused-vars
+      );
+  }
+
+  updateFilters(event) {
+    event.preventDefault();
+    this.setState({ inLoadingState: true });
+    this.updateSubjectClassifications();
+  }
+
+  doTheseVisitsMatchSlidingWindowCriteria(visitArray) {
+    // The length of the array input to this function should be
+    // == (this.state.numConsecutiveMonthsFromUser / 6)
+    const upperBound = visitArray.length;
+    for (let i = 0; i < upperBound; i += 1) {
+      const vloadCheck = visitArray[i].viral_load < this.state.viralLoadFromUser;
+      const therapyCheck = this.therapyValuesOfInterest.includes(visitArray[i].thrpyv);
+      if (!vloadCheck || !therapyCheck) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  classifySubjectPTC(subjectId, visitArray, thisContext) { // eslint-disable-line
+    const subjectWithVisits = { subject_id: subjectId,
+      consecutive_haart_treatments_begin_at_followup: 'N/A',
+      follow_ups: visitArray };
+
+    const slidingWindowSize = Math.ceil(thisContext.state.numConsecutiveMonthsFromUser / 6);
+    if (visitArray.length < slidingWindowSize + 1) {
+      thisContext.setState({ subjectNeither:
+                [...thisContext.state.subjectNeither, subjectWithVisits] });
+      return;
+    }
+
+    // The sliding window step. window is of size this.state.numConsecutiveMonthsFromUser
+    for (let i = 0; i < visitArray.length - slidingWindowSize; i += 1) {
```

Comment:
The algorithm will be faster if you do in reversed direction:
- Find the sliding window of interested therapy from the end. Let's call it `follow_ups[i, i + interested_length]`.
- If that sliding windows continue with a visit that has `thrpyv != interested therapy` and `viral_load  <= 4000`. Return that windows. If not, you can skip/jump to all sliding windows `follow_ups[h, i]` where `for all j`, such that `h <= j <=i, follow_ups[j] == interested_therapy`. Because, at then end of whatever the sliding windows, the post treatment condition will be violated.